#!/usr/bin/env python
PACKAGE = "plane_segment"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

sac_enum = gen.enum([ gen.const("SAC_RANSAC", int_t, 0, "SAC_RANSAC"),
			gen.const("SAC_LMEDS", int_t, 1, "SAC_LMEDS"),
			gen.const("SAC_MSAC", int_t, 2, "SAC_MSAC") ],
			"An enum to set SAC method type")

gen.add("ransac_method_type", int_t, 0, "", 0, 0, 2, edit_method=sac_enum)
gen.add("ransac_points_left_persentage", double_t, 0, "", 0.3, 0.01, 0.5)
gen.add("ransac_distance_threshold", double_t, 0, "", 0.03, 0, 0.1)
gen.add("ransac_max_iterations", int_t, 0, "", 100, 10, 1000)
gen.add("ransac_min_points_size", int_t, 0, "", 10000, 100, 10000)

exit(gen.generate(PACKAGE, "plane_segment", "RansacSegment"))

